docs: add documentation for multirm (#9016)
determined-ai/determined: docs/release-notes/multirm-for-k8s.rst

:orphan:

**New Features**

-  Kubernetes: Add ability to set up Determined master service on one Kubernetes cluster and

>    schedule workloads across the same or other Kubernetes clusters. Additional non-default resource

Reviewer: -  Kubernetes: Add ability to set up the Determined master service on one Kubernetes cluster and manage workloads across different Kubernetes clusters.

   managers and resource pools are configured under the master configuration options
   additional_resource_managers and resource_pools (additional resource managers are required to

>    have at least one resource pool defined). Additional resource managers and their resource pools

Reviewer: ``additional_resource_managers`` and ``resource_pools``

>    ``additional_resource_managers`` and ``resource_pools`` (additional resource managers are
>    required to have at least one resource pool defined). Additional resource managers and their
>    resource pools must have unique names.

Reviewer 2: ```suggestion
   ``additional_resource_managers`` and ``resource_pools``. (Add link to docs)
```

I think its fine to keep it high level and link to the docs

Author: @tara-det-ai are we linking to docs in release notes?

Reviewer: let's do both. humans scan release notes. searchers like headings. a little repetition can't hurt.

Add this to release notes: 

For more information, visit :ref:`master configuration <master-config-reference>`.

-  WebUI: Add ability to view resource manager name for resource pools.

-  API/CLI/WebUI: Route any requests to resource pools not defined in the master configuration to
   the default resource manager, not any additional resource manager, if defined.

-  Configuration: Add a ``name`` and ``metadata`` field to resource manager section in the master
   configuration. Add an ``additional_resource_managers`` section that follows the

>    ``resource_manager`` and ``resource_pool`` configuration pattern.

Author: Where else would we put them? I think each of these changes are important for the end user to know -- we make release notes regularly for config changes, and changes in API behavior.  We don't currently have a dedicated page to multiRM, so I thought they'd belong here

>    ``resource_manager`` and ``resource_pool`` configuration pattern.

Reviewer: I also like these updates in release notes for the reasons i posted in earlier comment. release notes are a great place to keep this info. if the users don't like it, they can skip it. it helps internally with cross referencing when we go back in later and check release notes against docs during doc reviews.
